### ISSUE TITLE
Not writting the path in the global attrs

### ISSUE THREAD

**Issue by juliettelavoie** [intake/intake-esm#470]
**Is your feature request related to a problem? Please describe.**

When calling `to_dataset_dict` on a single file, all the columns are written in the global attrs. Hence, I end up with the path in the dataset and I need to remove it manually before sharing my files publicly with that sensitive information. I personally don't see any benefits to having the path in the global attrs and I think most people would probably prefer to keep it private.

**Describe the solution you'd like**

I suggest removing the path column before adding the additional attributes to the dataset.
Maybe something like `additional_attrs=record.drop(self.path_column_name).to_dict(),` on l.221 of `source.py` ?

**Comment by juliettelavoie** [comment]
I am using the catalog as an internal tool for the creation of my dataset. When I am done, I am sharing the output files (with attributes), but not the catalog. In these public files, I don't want to keep the path from our internal servers. 

When I wrote the issue, I couldn't think of a use case where a user would need the path in the attrs, but there might be things I have not thought about. Since I wrote the  issue, I have just been deleting the path attribute by hand (similar to your suggestion here) and it's been fine.
